loadtest/control/actions.go

```diff
@@ -486,3 +486,145 @@ func ViewUser(u user.User) UserActionResponse {
 	}
 	return UserActionResponse{Info: fmt.Sprintf("viewed user %s", member.UserId)}
 }
+
+// Reload simulates the given user reloading the page
+// while connected to the server by executing the API
+// calls seen during a real page reload.
+func Reload(u user.User) UserActionResponse {
+	// Getting preferences.
+	err := u.GetPreferences()
+	if err != nil {
+		return UserActionResponse{Err: NewUserError(err)}
+	}
+
+	prefs, _ := u.Store().Preferences()
+	var userIds []string
+	chanId := ""
+	for _, p := range prefs {
+		switch {
+		case p.Name == model.PREFERENCE_NAME_LAST_CHANNEL:
+			chanId = p.Value
+		case p.Category == model.PREFERENCE_CATEGORY_DIRECT_CHANNEL_SHOW:
+			userIds = append(userIds, p.Name)
+		}
+	}
+
+	if chanId != "" {
+		// Marking the channel as viewed
+		_, err := u.ViewChannel(&model.ChannelView{
+			ChannelId:     chanId,
+			PrevChannelId: "",
+		})
+		if err != nil {
+			return UserActionResponse{Err: NewUserError(err)}
+		}
+	}
+
+	if ok, err := u.IsSysAdmin(); ok && err != nil {
+		err = u.GetConfig()
+		if err != nil {
+			return UserActionResponse{Err: NewUserError(err)}
+		}
+	}
+
+	err = u.GetClientLicense()
+	if err != nil {
+		return UserActionResponse{Err: NewUserError(err)}
+	}
+
+	// Getting the user.
+	userId, err := u.GetMe()
```

Comment:
We are not using `userId` yet, we can move it to L:562.

Reply:
Yes, a few things changed during the last refactor. We can probably move this. I just want to verify whether we were also trying to mimic the order of API calls. Obviously the webapp is mostly asynchronous but I some calls are always made before others so it would be wise to keep this code in sync as much as possible with what is happening on the client side.

Reply:
Sounds reasonable.